### PR TITLE
Feature/platfowner/new feature

### PR DESCRIPTION
**Pull request by platfowner**
Summary:
- Make SET operation atomic
- Fix temp state version conflict bug
- Make BATCH transaction results include tx hashes
- Use parseOrLog() instead of JSON.parse() to detect server-side errors easily
- Rename and move executeTransaciton() to BlockchainNode from P2PServer
- Move constants.js and chain-util.js to common/ directory 

#163

### CODE REVIEW

File: node/index.js [hunk 1]
```diff
@@ -232,6 +232,61 @@ class BlockchainNode {
...
+  /**
+   * Try to executes a transaction on the node database. If it was not successful, all changes are
+   * rolled back from the database states.
+   * @param {Object} tx transaction
+   */
+  executeOrRollbackTransaction(tx) {
+    const backupVersion = StateManager.createRandomVersion(`${StateVersions.BACKUP}`);
+    const backupRoot = this.stateManager.cloneVersion(this.db.stateVersion, backupVersion);
+    if (!backupRoot) {
+      logger.error(`[${LOG_HEADER}] Failed to clone state version: ${this.db.stateVersion}`);
```
Comment (liayoo) [inline node/index.js:244]: should we not execute the transaction in this case?

Comment (platfowner) [reply node/index.js:244]: I think that's reasonable. Just updated.

**Comment by platfowner** [comment]
@liayoo Comments addressed. PTAL

Also the following changes were added:
- Fix temp state version conflict bug
- Make BATCH transaction results include tx hashes
- Use parseOrLog() instead of JSON.parse() to detect server-side errors easily

File: db/state-manager.js [hunk 2]
```diff
@@ -165,18 +165,26 @@ class StateManager {
...
       logger.error(`[${LOG_HEADER}] Non-existing version: ${version}`);
       return false;
     }
-    this.finalizedVersion = version;
+    this.finalVersion = version;
     return true;
   }
+
+  /**
+   * Returns a random state version with the given previs.
```
Comment (liayoo) [inline db/state-manager.js:182]: minor) typo?

File: db/state-manager.js [hunk 3]
```diff
@@ -165,18 +165,26 @@ class StateManager {
...
+   * @param {*} versionPrefix state version prefix
+   */
+  static createRandomVersion(versionPrefix) {
+    return `${versionPrefix}:${Date.now()}:${Math.floor(Math.random()*10000)}`;
```
Comment (liayoo) [inline db/state-manager.js:186]: spaces between *

**Review by liayoo — APPROVED** [review]
LGTM with some minor comments. Thanks for the updates!

File: db/state-manager.js (hunk 3 above)
Comment (platfowner) [reply db/state-manager.js]: Fixed, thanks.

File: db/state-manager.js (hunk 2 above)
Comment (platfowner) [reply db/state-manager.js]: Fixed. Thanks!

**Pull request merged**